backends/test/suite/README.md

```diff
@@ -0,0 +1,15 @@
+# Backend Test Suite
+
+This directory contains tests that validate correctness and coverage of backends. These tests are written such that the backend is treated as a black box. The test logic verifies that the backend is able to handle a given pattern without erroring out (not partitioning is fine) and is able to run the graphs and yield reasonable outputs. As backends may differ significantly in implementation, numerical bounds are intentionally left loose.
```

Comment:
Add a line about why do we need this suite? 
And also for what does it mean to run this against a backend?
Lastly, how something ends up in a suite vs. not?

Reply:
I've added some additional wording here, clarifying that it's experimental and under active development. Let's chat about evolution of the test suite as a follow-up.